app/webapp/src/main/java/uk/ac/ebi/fg/annotare2/web/gwt/editor/client/view/experiment/design/AssignFilesDialog.java

```diff
@@ -87,19 +87,24 @@ public AssignFilesDialog(DialogCallback<Map.Entry<FileType, Integer>> callback,
             @Override
             public void onChange(ChangeEvent event) {
                 columnListHelp.setHTML(values.get(columnListBox.getSelectedIndex()).getHelpText());
-                if(experimentProfileType.isMicroarray() && values.get(columnListBox.getSelectedIndex()).isProcessed()){
-                    noOfColumnsLabel.setVisible(true);
-                    noOfColumns.setVisible(true);
+                if(experimentProfileType.isSequencing() || (experimentProfileType.isMicroarray() && values.get(columnListBox.getSelectedIndex()).isProcessed())){
+                    makeNoOfColumnsFieldVisible(true);
```

Comment:
We heard from Uncle Bob and I also did some googling and the result was the same: using a boolean parameter is a code smell.
Here is an article about it: https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3
I let you choose a decision about refactoring the code.
Thanks